spec/datadog/tracing/contrib/support/tracer_helpers.rb

```diff
@@ -92,6 +92,11 @@ def clear_traces!
       # with mock assertions.
       config.around do |example|
         example.run.tap do
+          traces = fetch_traces(tracer)
+          traces.each do |trace|
+            # write traces after the test to the agent in order to not mess up assertions
+            tracer.writer.write(trace)
+          end
```

Comment:
I'm not sure we want to execute this step all the time, since there might be cases where the agent or the test agent is not running

Reply:
Added a check for this